rancher-sandbox/rancher-desktop
# Automate deployment profile tests

Fixes #4963

These tests can be run twice, once with a system profile in place, and once without. Because the
system profile can be modified only with admin access, the test assumes it's not being run by
an admin, and skips any test that doesn't meet the expectation for the presence or absence of
a system profile.

## How to test

You'll need a way to manually create and delete system profiles to test the files.

The best way to test these quickly is:

Delete any system profile, and run
`yarn test:e2e e2e/startup-profiles.e2e.spec.ts` . All tests should either skip or pass.

Now create a system profile (and delete the user profile if it's incorrect).

Rerun `yarn test:e2e e2e/startup-profiles.e2e.spec.ts` .  All tests should either skip or pass. The tests that were skipped without a system profile should now pass, and vice versa. The key deciding factor should be the presence or absence of a system profile, since we can create and delete user-profiles and the settings file on all platforms.

## Scope

Note that the epic #4963 references epic #4962, which stipulates testing the preference hierarchy (e.g. user-defaults hides system-defaults). These tests only test for the first-run window (or lack thereof), because the bats tests can't test for this. Testing the preference hierarchy in this issue would be redundant.
